common/serde-reflection/src/ser.rs

```diff
@@ -4,18 +4,19 @@
 use crate::{
     error::{Error, Result},
     format::*,
-    trace::Tracer,
+    trace::{Records, Tracer},
     value::Value,
 };
 use serde::{ser, Serialize};
 
 pub(crate) struct Serializer<'a> {
     tracer: &'a mut Tracer,
+    records: &'a mut Records,
```

Comment:
Why don't you need a separate lifetime specifier here, but you do for deserialize?

Reply:
This is a very good question and I should probably add a comment to explain.

The lifetime 'a will keep shrinking as we create serializer objects in the calls to serialize internal structures. In the case of the Serializer, `tracer` and `records` are similar to the output string of a normal serializer, and that's totally fine.